### PR TITLE
#174797472 User role settings

### PR DESCRIPTION
**Pull request by elvisiraguha**
#### What does this PR do?
enable user role settings
#### Description of Task to be completed?
- create page for assigning roles to users
- create page for assigning permissions to roles
- create page for creating new roles
#### How should this be manually tested?
- Clone the repo git clone `https://github.com/atlp-rwanda/script-heroes-frontend.git`
- Switch your working directory `cd script-heroes-frontend`
- Checkout the branch git checkout `ft-user-role-settings-#174797472`
- Install all dependencies `yarn install`
- Start the dev server `yarn run dev`
- Visit roles page in the browser `http://localhost:8080/roles`
#### What are the relevant pivotal tracker stories?
[https://www.pivotaltracker.com/story/show/174797472](#174797472)
#### Screenshots (if appropriate)
Screen Shot 2020-11-30 at 13 47 20

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Good Job but try to move url into dotenv

File: redux/actions/roles.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const url = process.env.URL;
+const token = localStorage.getItem("token");
+
+const axiosBase = axios.create({
+  baseURL: url,
+  headers: {
+    "x-auth-token": token,
+    "Content-Type": "application/json",
+  },
+});
```
Comment (Habinezajanvier) [inline redux/actions/roles.js]: Shouldn't be good if you put this Axios functionality aside and you export them as they will be needed by everyone who needs to send an external request

File: redux/actions/users.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const token = localStorage.getItem("token");
+
+const axiosBase = axios.create({
+  baseURL: url,
+  headers: {
+    "x-auth-token": token,
+    "Content-Type": "application/json",
+  },
+});
+
```
Comment (Habinezajanvier) [inline redux/actions/users.js]: This can be written once and be re-used to prevent repeating yourself

File: redux/actions/roles.js (hunk 1 above)
Comment (elvisiraguha) [reply redux/actions/roles.js]: Fixed 👍

**Review by Habinezajanvier — APPROVED** [review]

**Review by ericus123 — APPROVED** [review]

File: src/views/Roles/Roles.js [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+          description: modalDescription,
+        })
+      );
+    }
+  };
+
+  const handleDelete = () => {
+    dispatch(deleteRole(roleInModal));
+  };
+
```
Comment (b0nbon1) [inline src/views/Roles/Roles.js]: this file so large please break it down, I could suggest you separate the logic from the views/components

**Review by b0nbon1 — CHANGES_REQUESTED** [review]

File: src/views/Roles/Roles.js (hunk 3 above)
Comment (elvisiraguha) [reply src/views/Roles/Roles.js]: Fixed 👍

**Review by b0nbon1 — APPROVED** [review]
Please rebase the new changes and it'll be ready to be merged

**Pull request merged**